### ISSUE TITLE
aws cloudformation validate-template parity

### ISSUE THREAD

**Issue by PatMyron** [aws-cloudformation/cfn-python-lint#1085]
Minimal template example that passes `cfn-lint` but fails `aws cloudformation validate-template`

```yaml
Resources:
  Type: AWS::S3::Bucket
```

`Template format error: [/Resources/Type] resource definition is malformed`

**Comment by kddejong** [comment]
Turns out we are checking for this but we had some issues when looking at directives that caused it to not get displayed.
